src/main/java/net/minecraftforge/client/render/BatchedBufferConfig.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.render;
+
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import org.lwjgl.opengl.GL11;
+
+public class BatchedBufferConfig
+{
+    public static final BatchedBufferConfig BLOCK = new BatchedBufferConfig(DefaultVertexFormats.BLOCK, GL11.GL_QUADS, BlockBufferDrawer.INSTANCE);
+
+    public final VertexFormat vertexFormat;
+    public final int glMode;
+    public final IBufferDrawer drawer;
+
+    public BatchedBufferConfig(VertexFormat vertexFormat, int glMode, IBufferDrawer drawer)
+    {
+        this.vertexFormat = vertexFormat;
+        this.glMode = glMode;
+        this.drawer = drawer;
+    }
+}
```

Comment:
implements eq/hash for situations where mods may use equivalent configs?

Reply:
Yeah I was thinking about that, but the problem is different mods may need a different GL state when rendering the format (like lighting enabled)

Reply:
Another problem here is that Vanilla MC doesn't allow for duplicate element types in vertex formats, except for UV. Even with Generic elements, IIRC, you're limited to one per format.  This sort of defeats the purpose of having custom vertex formats.  I patch VertexFormatElement.isFirstOrUV() to allow more than one generic element in the core mod I mentioned elsewhere.  You'll probably want to do something similar as part of this pull request.  (And then I won't need to . :-)

Reply:
@grondag The purpose of this is to allow existing rendering methods to be more performant. If you think this is a problem make your own PR

Reply:
Sorry, I was initially confused by the title, which doesn't mention performance.